conversion.c

```diff
@@ -3,28 +3,25 @@
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
 // conversion.c
-// Program to convert decimal number into binary and hexadecimal 
+// Program to convert a decimal number to binary and hexadecimal.
 // ------------------------------------------------------------------------------------------------
 
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdbool.h>
 #include <malloc.h>
+#include <limits.h>
 #pragma warning (disable:4996)
 #define ERROR_MEM_ALLOC_FAILURE -1
 #define ERROR_EXCEEDED_LENGTH -2
 #define ERROR_INVALID_INPUT -3
 
-
 char Arr[16] = { '0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F' };    // Array of hexadecimal equivalent characters
 
 /// <summary>Returns the nearest byte length</summary>
 int NearestByte (int digit, bool isLargestNum) {

```

Comment:
Good idea to check and use the required number of bits to represent a number instead of representing everything as a 32 bit number